app/main/views/login.py

```diff
@@ -0,0 +1,355 @@
+    else:
+        template_data = main.config['BASE_TEMPLATE_DATA']
+        return render_template("auth/reset-password.html",
+                               email_address=email_address,
+                               form=form,
+                               token=token,
+                               **template_data), 400
+
+
+@main.route('/create-account', methods=["GET"])
+def create_buyer_account():
+    form = EmailAddressForm()
+    template_data = main.config['BASE_TEMPLATE_DATA']
+
+    return render_template(
+        "auth/create-buyer-account.html",
+        form=form,
+        **template_data), 200
+
+
+@main.route('/create-account', methods=['POST'])
+def submit_create_buyer_account():
+    current_app.logger.info(
+        "buyercreate: post create-buyer-account")
+    template_data = main.config['BASE_TEMPLATE_DATA']
+    form = EmailAddressForm()
+
+    if form.validate_on_submit():
+        email_address = form.email_address.data
+        # Send email then
+        # Go to "ACTIVATE YOUR ACCOUNT" page
+        token = generate_token(
+            {
+                "email_address":  email_address
+            },
+            current_app.config['SHARED_EMAIL_KEY'],
+            current_app.config['INVITE_EMAIL_SALT']
+        )
+        url = url_for('main.create_user', encoded_token=token, _external=True)
+        email_body = render_template("emails/create_buyer_user_email.html", url=url)
+        try:
+            send_email(
+                email_address,
+                email_body,
+                current_app.config['DM_MANDRILL_API_KEY'],
+                current_app.config['CREATE_USER_SUBJECT'],
+                current_app.config['RESET_PASSWORD_EMAIL_FROM'],
+                current_app.config['RESET_PASSWORD_EMAIL_NAME'],
+                ["user-creation"]
+            )
+            session['email_sent_to'] = email_address
+        except MandrillException as e:
+            current_app.logger.error(
+                "buyercreate.fail: Create user email failed to send. "
+                "error {error} email_hash {email_hash}",
+                extra={
+                    'error': six.text_type(e),
+                    'email_hash': hash_email(email_address)})
+            abort(503, "Failed to send user creation email")
+
+        data_api_client.create_audit_event(
+            audit_type=AuditTypes.invite_user,
+            data={'invitedEmail': email_address})
+
+        return redirect(url_for('.create_your_account_complete'), 302)
+    else:
+        return render_template(
+            "auth/create-buyer-account.html",
+            form=form,
+            email_address=form.email_address.data,
+            **template_data
+        ), 400
+
+
+@main.route('/create-user/<string:encoded_token>', methods=["GET"])
+def create_user(encoded_token):
+    form = CreateUserForm()
+    template_data = main.config['BASE_TEMPLATE_DATA']
+
+    token = decode_invitation_token(encoded_token, buyer=True)
```

Comment:
In the apiclient we have a supplier flag, here we have a buyer flag. I wonder if both of these would make more sense as a role argument.